### PR TITLE
refactor stocking batches

### LINKED ISSUE #164: refactor stocked batch with more appropriate approach
As of now, there is a step type called STOCKED.
This is probably not the right way to represent a stocked batch for two reasons:

1) stocking is NOT a production step, so should not be treated this way.
2) since it is now a step type, you can actually move from FINALIZED to STOCKED in the "move to next step" functionality without actually stocking the batch in the warehouse.

For this reasons, i'm planning a small refactor for representing a stocked batch.

### PR DESCRIPTION
**Pull request by GiackAloZ**
fixes #164

### CODE REVIEW

**Comment by GiackAloZ** [comment]
Thinking about this refactor, because actually it could be better to have stocked as a step, not sure yet

**Comment by GiackAloZ** [comment]
i'm reopening this because i still think it is a good refactor and it makes sense to use it this way

File: src/main/java/nwoolcan/model/brewery/batch/BatchImpl.java [hunk 1]
```diff
@@ -145,13 +148,38 @@ public boolean isEnded() {
...
+
+    @Override
+    public Result<Empty> stockBatchInto(final Stock stock) {
+        return Result.of(stock)
+                     .require(this::isEnded)
+                     .require(() -> !this.isStocked(), new IllegalStateException())
+                     .require(s -> s.getArticle().getUnitOfMeasure().equals(this.getCurrentSize().getUnitOfMeasure()))
+                     .peek(this::setStockReference)
+                     .toEmpty();
+    }
```
Comment (samueleceroni) [inline src/main/java/nwoolcan/model/brewery/batch/BatchImpl.java]: I think you are checking a lot of thing two times when it is not needed and that the order of calls are not ok. what if some of these require do not pass and you have already created the stock and added a record there? Refer to the other comment for more details on how i would swap method calls.

File: src/main/java/nwoolcan/model/brewery/BreweryImpl.java [hunk 2]
```diff
@@ -80,14 +83,15 @@ public BatchBuilder getBatchBuilder() {
...
-                     .toEmpty();
+                     .require(Batch::isEnded, new IllegalStateException(CANNOT_STOCK_NOT_ENDED_BATCH_MESSGE))
+                     .require(b -> !b.isStocked(), new IllegalStateException(ALREADY_STOCKED_BATCH_MESSAGE))
+                     .require(b -> b.getCurrentSize().getUnitOfMeasure().equals(beerArticle.getUnitOfMeasure()), new IllegalArgumentException(WRONG_ARTICLE_UNIT_OF_MEASURE_MESSAGE))
+                     .flatMap(b -> createBeerStock(beerArticle, expirationDate, b))
+                     .flatMap(s -> {
+                         return s.addRecord(new Record(batch.getCurrentSize(), Record.Action.ADDING))
+                                 .map(e -> s);
+                     })
+                     .flatMap(batch::stockBatchInto);
```
Comment (samueleceroni) [inline src/main/java/nwoolcan/model/brewery/BreweryImpl.java]: I think you should before set the stock reference in the batch, and then add the record in the beerstock.

**Comment by GiackAloZ** [comment]
i will close this for now, reopen it later when me and @samueleceroni found a way to solve this problem

**Pull request closed without merging**